test/common/http/conn_manager_impl_test.cc

```diff
@@ -1039,6 +1094,9 @@ TEST_F(HttpConnectionManagerImplTest, MultipleFilters) {
 
   setupFilterChain(3, 2);
 
+  // Test route caching.
```

Comment:
In conn_manager_impl.cc, when checking for websocket upgrade compatibility, we are fetching the routes and checking for the "allow_websocket_upgrade" flag. Since websocket introduces a call to route() way before entering any route filter, this EXPECT_CALL has to be moved before the expects for filters. Otherwise, the code segfaults. :(